core/storage/question/gae_models.py

```diff
@@ -372,13 +376,18 @@ def get_question_skill_links_equidistributed_by_skill(
                 given skill ids, but the order of questions for the same skill
                 is random.
         """
+        if len(skill_ids) > feconf.MAX_NUMBER_OF_SKILL_IDS:
+            raise Exception('Please keep the number of skill IDs below 20.')
+
         question_count_per_skill = int(
             math.ceil(float(total_question_count) / float(len(skill_ids))))
         question_skill_link_models = []
         existing_question_ids = []
 
         for skill_id in skill_ids:
             query = cls.query(cls.skill_id == skill_id)
+            # The number of questions fetched will be dropped too low because
```

Comment:
"We fetch more questions here in order to try and ensure that the eventual number of returned questions is sufficient to meet the number requested, even after de-duplication."

Reply:
Done.